### PR TITLE
[GOBBLIN-1027] add metrics for users running gaas jobs

### PR DESCRIPTION
**Pull request by arjun4084346**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below! @sv2000 , Pawan , please review

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1027

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
it adds a counter to monitor the number of jobs submitted by a user

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
trivial changes

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 1]
```diff
@@ -788,6 +794,33 @@ private ContextAwareCounter getRunningJobsCounter(DagNode<JobExecutionPlan> dagN
               dagNode.getValue().getSpecExecutor().getUri().toString()));
     }
 
+    private List<ContextAwareCounter> getRunningJobsCounterForUser(DagNode<JobExecutionPlan> dagNode) {
+      Properties props = dagNode.getValue().getJobSpec().getConfigAsProperties();
```
[the remark this answers is not in the document]
Comment (arjun4084346) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: No reason. What's the advantage of using Config?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 2]
```diff
@@ -788,6 +794,33 @@ private ContextAwareCounter getRunningJobsCounter(DagNode<JobExecutionPlan> dagN
...
+      String proxy = props.getProperty("user.to.proxy");
```
[the remark this answers is not in the document]
Comment (arjun4084346) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: have a null check in next line for this reason.

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sv2000) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: I think we should as a rule use typesafe config unless usage of Java properties is necessary (e.g. for Azkaban AbstractJob or maintaining backwards compatibility). Typesafe config has many advantages over Properties. In gobblin-service, we largely use typesafe config. From the point of view of maintaiining code uniformity, I think usage of config is desirable. We also have developed a bunch of utility methods around typesafe config in Gobblin ConfigUtils that has made the usage of types,

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 3]
```diff
@@ -796,7 +796,7 @@ private ContextAwareCounter getRunningJobsCounter(DagNode<JobExecutionPlan> dagN
 
     private List<ContextAwareCounter> getRunningJobsCounterForUser(DagNode<JobExecutionPlan> dagNode) {
       Properties props = dagNode.getValue().getJobSpec().getConfigAsProperties();
-      String proxy = props.getProperty("user.to.proxy");
+      String proxy = props.getProperty(AzkabanProjectConfig.USER_TO_PROXY);
       List<ContextAwareCounter> counters = new ArrayList<>();
 
       if (proxy != null) {
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: Nit: Can this be replaced with if (!Strings.isNullOrEmpty(proxy)) similar to what you have in line 811?

**Pull request closed without merging**